src/games/rota/game.ts

```diff
@@ -0,0 +1,166 @@
+}
+
+function isVictory(G: IG): boolean {
+  for (const p of G.points) {
+    if (p.playerID !== null) {
+      const axisVar: number[] = [-1, 0, 1];
+      // check along x-axis
+      let testWin = true;
+      for (const y of axisVar) {
+        testWin = testWin && p.playerID === findPoint(G, p.x, y).playerID;
+      }
+      if (testWin) {
+        return true;
+      }
+
+      // check along x-axis
+      testWin = true;
+      for (const x of axisVar) {
+        testWin = testWin && p.playerID === findPoint(G, x, p.y).playerID;
+      }
+      if (testWin) {
+        return true;
+      }
+
+      // check Corners Points
+      if (p.x !== 0 && p.y !== 0) {
+        // test around the circumference
+        testWin = p.playerID === findPoint(G, 0, p.y).playerID;
+        testWin = testWin && p.playerID === findPoint(G, p.x, 0).playerID;
+        if (testWin) {
+          return true;
+        }
+
+        // test diagonal
+        testWin = p.playerID === findPoint(G, 0, 0).playerID;
+        testWin = testWin && p.playerID === findPoint(G, -p.x, -p.y).playerID;
+        if (testWin) {
+          return true;
+        }
+      }
+    }
+  }
+  return false;
+}
+
+function placePiece(G: IG, ctx: IGameCtx, pointID: number) {
+  // if more than 6 pieces are placed, do nothing
+  let piecesPlaced = G.piecesPlaced;
+  if (piecesPlaced >= 6) {
+    return { ...G };
+  }
+
+  // add player-id of player who clicked the point
+  const points = G.points.map((point, idx) => {
+    if (point.playerID === null && pointID === idx) {
+      piecesPlaced = piecesPlaced + 1;
+      return { ...point, playerID: ctx.currentPlayer, pieceID: G.piecesPlaced };
+    } else {
+      return { ...point };
+    }
+  });
+
+  return { ...G, points, piecesPlaced };
+}
+
+export function movePiece(G: IG, ctx: IGameCtx, currentID: number, newID: number): IG | string {
+  if (
+    G.points[currentID].playerID === null ||
+    G.points[newID].playerID !== null ||
+    G.points[currentID].playerID !== ctx.playerID // Check if player owns this piece
+  ) {
+    return INVALID_MOVE;
+  }
+
+  // check if the move is valid (by checking if distance is greater than 1.5)
+  const n1 = 1 / (Math.sqrt(G.points[currentID].x ** 2 + G.points[currentID].y ** 2) || 1);
+  const n2 = 1 / (Math.sqrt(G.points[newID].x ** 2 + G.points[newID].y ** 2) || 1);
+  const distance = Math.sqrt(
+    (G.points[currentID].x * n1 - G.points[newID].x * n2) ** 2 +
+      (G.points[currentID].y * n1 - G.points[newID].y * n2) ** 2,
+  );
+  if (distance > 1.1) {
+    return INVALID_MOVE;
+  }
+
+  // update the state
+  const points = G.points.map((p, idx) => {
+    if (idx === currentID) {
+      return { ...p, playerID: null, pieceID: null };
+    } else if (idx === newID) {
+      return { ...p, playerID: G.points[currentID].playerID, pieceID: G.points[currentID].pieceID };
+    } else {
+      return { ...p };
+    }
+  });
+
+  return { ...G, points };
+}
+
+const GameConfig: IGameArgs = {
```

Comment:
Please add tests, these are the most important tests because we need to make sure the rules are correct (now and in the future when we upgrade dependencies)